File: src/main/java/forpleuvoir/hiirosakura/client/compat/modmenu/ModMenuImpl.java
package forpleuvoir.hiirosakura.client.compat.modmenu;

import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;
import forpleuvoir.hiirosakura.client.gui.GuiConfig;

/**
 * ModMenu 接口
 *
 * @author forpleuvoir
 * <p>#project_name hiirosakura
 * <p>#package forpleuvoir.hiirosakura.client.compat.modmenu
 * <p>#class_name ModMenuImpl
 * <p>#create_time 2021/6/10 21:50
 */
public class ModMenuImpl implements ModMenuApi {

    @Override
    public ConfigScreenFactory<?> getModConfigScreenFactory() {
        return (screen) -> {
            var gui = new GuiConfig();
            gui.setParent(screen);
            return gui;
        };
    }
}

File: src/main/kotlin/forpleuvoir/hiirosakura/client/compat/modmenu/ModMenuImpl.kt
package forpleuvoir.hiirosakura.client.compat.modmenu

import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import forpleuvoir.hiirosakura.client.gui.HiiroSakuraScreen

/**
 * ModMenu 接口
 *
 * @author forpleuvoir
 *
 * 项目名 hiirosakura
 *
 * 包名 forpleuvoir.hiirosakura.client.compat.modmenu
 *
 * 文件名 ModMenuImpl
 *
 * 创建时间 2021/6/10 21:50
 */
class ModMenuImpl : ModMenuApi {
	override fun getModConfigScreenFactory(): ConfigScreenFactory<*> {
		return ConfigScreenFactory {
			return@ConfigScreenFactory HiiroSakuraScreen.current().apply { parent = it }
		}
	}
}